[#12] [UI] As a signed-in user, I can see the Home Screen

ducbm051291 opened the pull request:
- #12 

## What happened

- Allow user to see the Home screen after signing in
 
## Insight

- Create HomeViewModel
- Create HomeSurveyItemView
- Create HomeHeaderView
- Create PageControlView
- Create HomeSkeletonLoadingView
- Create HomeLeftMenuView
- Create Modifiers
- Update Font sizes, Localizable strings, and Colors
- Update Navigator with embed in navigation view parameter
- Add ShimmerView and Kingfisher library to Podfile
 
## Proof Of Work

https://user-images.githubusercontent.com/24598204/207531561-711a6f77-323e-4707-bbd0-d9343ac1feb4.mov

blyscuit commented:
This is many views in one PR, it's ideal to separate then to multiple PR for ease of review.

ducbm051291 commented on Surveys/Sources/Presentation/Modules/HomeView/View/HomeLeftMenuView.swift:
```diff
@@ -0,0 +1,126 @@
+        VStack(alignment: .leading) {
+            Spacer()
+            HStack {
+                Text(Bundle.main.version)
```
Agree

ducbm051291 commented on Surveys/Sources/Presentation/Modules/HomeView/HomeView.swift:
```diff
@@ -13,10 +13,102 @@ import UIKit
+                    }
+                    .overlay(alignment: .top) {
+                        if isMenuVisible {
+                            HomeLeftMenuView(isVisible: $isMenuVisible)
```
You caught me. You're right, I was totally wrong here. It should be name HomeRightMenuView 😢

ducbm051291 commented on Surveys/Sources/Domain/Entities/Survey.swift:
```diff
@@ -0,0 +1,14 @@
+
+import Foundation
+
+protocol Survey: Decodable {
```
Yes, this is for the backend task after this

blyscuit replied:
In that case you will create another object for the data layer which is Decodable, there would be no case for this protocol to be `Decodable` is there 🤔

ducbm051291 commented:
> This is many views in one PR, it's ideal to separate then to multiple PR for ease of review.

I've updated the code and removed Skeleton Loading View and Menu View for implementation in another story

blyscuit commented on Surveys/Sources/Presentation/Modules/HomeView/View/HomeHeaderView.swift:
```diff
@@ -0,0 +1,52 @@
+        }, label: {
+            KFImage(URL(string: imageURL))
+                .placeholder { _ in
+                    Assets.userPlaceholderIcon.image
```
I think we can use Color.black for this. No need to add an extra image. When the img is being load from the url, users will see this placeholder image changing to the url image which might not be the best experience, what do you think 🤔

Shayokh144 commented on Surveys/Sources/Presentation/Modules/HomeView/HomeView.swift:
```diff
@@ -13,10 +13,97 @@ import UIKit
+    private func setUpPageControl() -> some View {
+        VStack(alignment: .leading) {
+            Spacer()
+            PageControlView(
```
@ducbm051291 if number of pages is big like 15 or 20 will it fit in the screen with current size?

ducbm051291 replied:
Yes, because behind this PageControlView is UIKit PageControl itself 😁 

Uploading Simulator Screen Recording - iPhone 14 Pro Max - 2022-12-15 at 13.20.24.mp4…

Shayokh144 replied:
@ducbm051291 the ui looks a bit wired as all the dots are not having same size, is it ok?

ducbm051291 replied:
@Shayokh144  I think that's ok because it's Apple default control
